allennlp/interpret/influence_interpreters/influence_interpreter.py

```diff
@@ -0,0 +1,361 @@
+from os import PathLike
+import re
+from typing import List, Optional, NamedTuple, Sequence, Union, Dict, Any
+
+import torch
+from torch import autograd
+
+from allennlp.common import Registrable, Lazy, plugins
+from allennlp.common.tqdm import Tqdm
+from allennlp.common.util import int_to_device
+from allennlp.data import Instance, DatasetReader, DatasetReaderInput, Batch
+from allennlp.data.data_loaders import DataLoader, SimpleDataLoader
+from allennlp.models import Model, Archive, load_archive
+from allennlp.nn.util import move_to_device
+
+
+class TrainInstanceInfluence(NamedTuple):
+    instance: Instance
+
+    loss: float
+
+    score: float
+    """
+    The influence score associated with this training instance.
+    """
+
+
+class InterpretOutput(NamedTuple):
+    test_instance: Instance
+
+    loss: float
+    """
+    The loss corresponding to the `test_instance`.
+    """
+
+    top_k: List[TrainInstanceInfluence]
+    """
+    The top `k` most influential training instances along with their influence score.
+    """
+
+    bottom_k: List[TrainInstanceInfluence]
+    """
+    The bottom `k` least influential training instances along with their influence score.
+    """
+
+
+class TrainInstance(NamedTuple):
+    """
+    Wraps a training `Instance` along with its associated loss and gradients.
+
+    `InfluenceInterpreter.train_instances` is a list of these objects.
+    """
+
+    instance: Instance
+    loss: float
+    grads: Sequence[torch.Tensor]
+
+
+class InfluenceInterpreter(Registrable):
+    """
+    A `SaliencyInterpreter` interprets an AllenNLP Predictor's outputs by assigning an influence
+    score to each training instance with respect to each test input.
+
+    Subclasses are required to implement the `calculate_influence_scores()` method.
+
+    # Parameters
+
+    model : `Model`, required
```

Comment:
Up above you said that this was for predictors, but here's it's a model.

Reply:
Yes good point, I'll remove mention of predictors.